OtripleS.Portal.Web.Tests.Unit/Services/Courses/CourseServiceTests.Exceptions.RetrieveAll.cs

```diff
@@ -0,0 +1,130 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE AS LONG AS SOFTWARE FUNDS ARE DONATED TO THE POOR
+// ---------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Moq;
+using OtripleS.Portal.Web.Models.Courses;
+using OtripleS.Portal.Web.Models.Courses.Exceptions;
+using Xunit;
+
+namespace OtripleS.Portal.Web.Tests.Unit.Services.Courses
+{
+    public partial class CourseServiceTests
+    {
+        [Theory]
+        [MemberData(nameof(CriticalApiExceptions))]
+        public async Task ShouldThrowCriticalDependencyExceptionOnRetrieveAllIfCriticialErrorOccursAndLogItAsync(
+            Exception criticalDependencyException)
+        {
+            // given
+            var failedCourseDependencyException =
+                new FailedCourseDependencyException(
+                    criticalDependencyException);
+
+            var expectedCourseDependencyException =
+                new CourseDependencyException(
+                    failedCourseDependencyException);
+
+            this.apiBrokerMock.Setup(broker =>
+                broker.GetAllCoursesAsync())
+                    .ThrowsAsync(criticalDependencyException);
+
+            // when
+            ValueTask<List<Course>> retrievedAllCoursesTask =
```

Comment:
`retrieveAllCoursesTask` instead of `retrievedAllCoursesTask ` since it's a task not a result which already has retrieved courses.

Reply:
On other places too

Reply:
@mtoncic  Hi now i'm confused. Only for the line 37 or all of the test cases ?. All of them are not a result, but i should have changed them before to retrieved. Thx in advane

Reply:
a ok that makes sence ;)

Reply:
@mtoncic done. thx